Workiva/w_dio
# FEDX-151 Finish setting up json_annotation and json_serializable copies, support analyzer 1+2

## Motivation
All versions of json_serializable that support analyzer 2.x generate null-safe code, which makes it more complicated to upgrade to analyzer 2.x in packages that aren't yet migrated to null safety.

Ideally, we'd like a version of json_serializable that resolves to analyzer 2.x but doesn't emit null-safe code.

This repo already has a copy of the last json_serializable version (3.5.2) that generated non-null-safe code, as well a copy of the associated json_annotation. However, these copies weren't fully set up, and needed to be upgraded to newer analyzer versions.

## Changes
- Finish setting up these json_ packages
    - Set up CI checks
    - Update references to the original packages to their copied versions. This includes package imports, and keys used in builders and their configurations
        - Update `json_serializable` references to `json_serializable_3_5_2`
        - Update `json_annotation` references to `json_annotation_3_1_1`
- Update json_serializable to support analyzer 1.x and 2.x. This included:
    - Updating dependencies
    - Backpatching some changes from newer json_serializable versions (see commit messages for original changes)
        - 9e0d05d56798fa7f045f53a6125c996de092999a
        - e2235d63a76509d3633ed347f281dc0a5725fd26

## QA instructions
- CI passes
- Validate CI setup:
    - json_serializable_quality_checks_analyzer_1 resolves to analyzer 1.x
    - json_serializable_quality_checks_analyzer_2 resolves to analyzer 2.x
    - json_serializable_and_json_annotation_integration resolves to overridden json_annotation path
- Verify there are no Pub publish errors (warnings are okay) when running `dart pub publish --dry-run` within the json_serializable-3.5.2 directory
